packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -287,17 +287,24 @@ export const usePopover = (options: PopoverOptions & PopoverInternalProps) => {
                 maxInlineSize: `min(${referenceWidth * 2}px, 90vw)`,
               }
             : {}),
+          ...(middleware.hide && {
+            visibility: floating.middlewareData.hide?.referenceHidden
+              ? 'hidden'
+              : 'visible',
+          }),
```

Comment:
We shouldn't set `visibility: visible`; we should just *do nothing* when `referenceHidden` doesn't match.

e.g. something like:

```js
...(middleware.hide
    && floating.middlewareData.hide?.referenceHidden && {
      visibility: 'hidden'
   })
```